app/script_variant.py

```diff
@@ -102,13 +113,17 @@ def romanize_word(word):
     if not word:
         return word
 
+    yomigana_list = []
     if has_kanji(word):
-        word = romanize_japanese_name_by_name_dict(word)
-        word = romanize_japanese_location(word)
+        yomigana_list = romanize_japanese_name_by_name_dict(word)
+        if not yomigana_list:
+            yomigana_list = romanize_japanese_location(word)
+        return ' '.join(word for word in yomigana_list)
```

Comment:
As discussed, return an array which contains (possibly multiple) Japanese readings and the romanization by unidecode here, and perform OR search in full_text_search.